server/controllers/finance/purchases.js

```diff
@@ -361,58 +362,82 @@ function find(options) {
+  const PURCHASE_FULLY_RECEIVED = 3;
+  const PURCHASE_EXCESSIVELY_RECEIVED = 6;
+
+  const glb = {};
   const purchaseUuid = db.bid(req.params.uuid);
-  const sql = `
-    SELECT IFNULL(SUM(m.quantity * m.unit_cost), 0) AS movement_cost, p.cost, CEIL(DATEDIFF(m.date, p.date)) AS delay
+
+  const sqlQuantities = `
+    SELECT SUM(t.purchase_quantity) AS purchase_quantity, SUM(t.movement_quantity) AS movement_quantity
+    FROM (
+      (
+        SELECT IFNULL(SUM(pi.quantity), 0) AS purchase_quantity, 0 AS movement_quantity FROM purchase p
+        JOIN purchase_item pi ON pi.purchase_uuid = p.uuid
+        WHERE p.uuid = ?
+      ) UNION (
+        SELECT 0 AS purchase_quantity, SUM(m.quantity) AS movement_quantity FROM stock_movement m 
+        JOIN lot l ON l.uuid = m.lot_uuid 
+        JOIN purchase p ON p.uuid = l.origin_uuid
+        WHERE p.uuid = ? AND m.flux_id = ? AND m.is_exit = 0
+      )
+    )t
+  `;
+
+  const sqlPurchaseDelay = `
+    SELECT
+      p.cost, CEIL(DATEDIFF(m.date, p.date)) AS delay
     FROM stock_movement m
     JOIN lot l ON l.uuid = m.lot_uuid
     JOIN purchase p ON p.uuid = l.origin_uuid
-    WHERE p.uuid = ? AND m.flux_id = ? AND m.is_exit = 0;
+    WHERE p.uuid = ? AND m.flux_id = ? AND m.is_exit = 0 LIMIT 1;
   `;
 
-  db.one(sql, [purchaseUuid, FROM_PURCHASE_ID])
-    .then((row) => {
-      let query = '';
-      status.cost = row.cost;
-      status.movement_cost = row.movement_cost;
-      status.delay = Math.round((row.delay / 30) * 10) / 10;
-
-      if (row.movement_cost === row.cost) {
-      // the purchase is totally delivered
-        status.status = 'full_entry';
-        query = 'UPDATE purchase SET status_id = 3 WHERE uuid = ?';
-      } else if (row.movement_cost > 0 && row.movement_cost < row.cost) {
-      // the purchase is partially delivered
-        status.status = 'partial_entry';
-        query = 'UPDATE purchase SET status_id = 4 WHERE uuid = ?';
-      } else if (row.movement_cost === 0) {
-      // the purchase is not yet delivered
-        status.status = 'no_entry';
-        // If there are no movements, the status of the purchase order must remain confirmed
-        query = 'UPDATE purchase SET status_id = 2 WHERE uuid = ?';
+  const sqlPurchaseInventories = `
+    SELECT purchase_item.inventory_uuid AS inventory_uuid, inventory.delay, inventory.num_delivery
+    FROM purchase_item
+    JOIN inventory ON inventory.uuid = purchase_item.inventory_uuid
+    WHERE purchase_item.purchase_uuid = ?
+  `;
+
+  const dbPromise = [
+    db.one(sqlQuantities, [purchaseUuid, purchaseUuid, FROM_PURCHASE_ID]),
+    db.one(sqlPurchaseDelay, [purchaseUuid, FROM_PURCHASE_ID]),
+    db.exec(sqlPurchaseInventories, [purchaseUuid]),
+  ];
+
+  q.all(dbPromise)
+    .spread((resultQuantities, resultDelay, resultPurchaseInventories) => {
+      const rq = resultQuantities;
+      const rd = resultDelay;
+
+      glb.delay = Math.round((rd.delay / 30) * 10) / 10;
```

Comment:
Where did these magic numbers come from?  Are you converting months -> days?  Do you really need the ` * 10) / 10`?

Could you put a comment above this line explaining it?

Reply:
They are magic !!!